onboarding.md

```diff
@@ -0,0 +1,63 @@
+
+#### Services and models 
+
+A `Service` is a module which exports functions that encapsulate CRUD-operations on an entity (like Organisation or User). Most of the time, a service should be paired with a `Model`. In it's core a `Model` is basically a `Joi-schema` that describes a particular entity, eg. an Organisation or an User. Joi-schemas are immutable, and can be "extended" using the [append() or keys()](https://joi.dev/api/?v=17.3.0#objectappendschema) method, so we have a [Default](./server/src/models/v2/Default.js)-model that describes properties shared by all entities. 
+
+All service-functions take a `knex`-parameters as the final parameter. This can be either a bound `knex`-instance, or a `transaction`-object from `knex.transaction()`, as mentioned above these can be used interchangeably. A service-function should always take and return internal data-structures, it's up to the function to transform the results from a query to interal structure. These functions are pretty similar to the old architecture's `data`-functions. However they are a bit more structured and composable. They are also easily integrated as [service-methods](https://hapi.dev/api/?v=20.0.1#server.methods), and thus cacheable.
+
+The `Model`-module exports Joi-schema definitions, along with functions `parseDatabaseJson` and `formatDatabaseJson`. The idea is that we have at least two definitions; database-definition and internal-definition. We could also have a third one for external schemas, but this should be mostly the same as the internal one as there is no need to complicate it further. These definitions can be used in any [hapi-validate](https://hapi.dev/api/?v=20.0.1#-routeoptionsvalidate)-function. The idea is that you use all the Joi-goodness for usecases like restricting and renames of keys. For instance `definition.without('sensitive')`. `Joi.alter()` can be used to group such transformations for a particular alteration, see [tailor()](https://joi.dev/api/?v=17.3.0#anytailortargets). 
+
+`parseDatabaseJson()` is meant to be a function that can be overriden for more specific transformations, and will be called with the result of a database-query and is meant to return the data for internal application use. `formatDatabaseJson` is the opposite - it takes data from the application and transforms it to database-friendly objects.
+
+
+Many would think that services and the models should be classes, and I've been a bit back and forth. I'm deliberately not using classes unless it's needed, and most of the time that means when you need state. Up to now these do not need state, but it could make sense to have a base-class that lays out the structure of these modules. But classes needs to instantiated, and that would mean bootstrapping the services on startup which is annoying in itself, and the methods would also not be statically available. [Schmervice](https://github.com/hapipal/schmervice) is also something we could consider.
```

Comment:
Should this paragraph be left out? I've multiple reflections like this that I want to get out and would like feedback on, but I'm unsure if they are just bloat here.

Reply:
I think it's fine, we can always clean it up and remove info later!